lib/rspec/support.rb

```diff
@@ -90,15 +90,18 @@ def self.class_of(object)
       singleton_class.ancestors.find { |ancestor| !ancestor.equal?(singleton_class) }
     end
 
-    # A single thread local variable so we don't excessively pollute that namespace.
+    # Stash original methods to allow the user to mock them.
     if RUBY_VERSION.to_f >= 2
-      def self.thread_local_data
-        Thread.current.thread_variable_get(:__rspec) || Thread.current.thread_variable_set(:__rspec, {})
-      end
+      @thread_variable_get = Thread.instance_method(:thread_variable_get)
+      @thread_variable_set = Thread.instance_method(:thread_variable_set)
```

Comment:
Typically we've used a constant for these sort of stashing.
```suggestion
      THREAD_SET = Thread.instance_method(:thread_variable_set)
```

Reply:
> Typically we've used a constant for these sort of stashing.

Done.

Reply:
Can you please make sure you’ve pushed it?

Reply:
Oops, I forgot to stash (no pun intended!) the changes before amending and pushing.